iceoryx_examples/waitset/README.md

```diff
@@ -398,43 +442,69 @@ The event loop calls `auto notificationVector = waitset.wait()` in a blocking ca
 receive a vector of all the _NotificationInfos_ which are corresponding to the occurred events.
 If the _Event_ originated from the `shutdownTrigger` we terminate the program.
 
+<!--[geoffrey][iceoryx_examples/waitset/ice_waitset_grouping.cpp][[event loop][shutdown path]]-->
 ```cpp
-while (true)
+while (keepRunning)
 {
     auto notificationVector = waitset.wait();
 
     for (auto& notification : notificationVector)
     {
         if (notification->doesOriginateFrom(&shutdownTrigger))
         {
-            return (EXIT_SUCCESS);
+            keepRunning = false;
         }
+        // we print the received data for the first group
+        // ...
+    }
+
+    std::cout << std::endl;
+}
 ```
 
 The remaining part of the loop is handling the subscribers. In the first group
 we would like to print the received data to the console and in the second group
 we just dismiss the received data.
 
+<!--[geoffrey][iceoryx_examples/waitset/ice_waitset_grouping.cpp][[event loop][data path]]-->
 ```cpp
-    else if (notification->getNotificationId() == FIRST_GROUP_ID)
-    {
-        auto subscriber = notification->getOrigin<iox::popo::UntypedSubscriber>();
-        subscriber->take().and_then([&](iox::popo::Sample<const void>& sample) {
-            const CounterTopic* data = reinterpret_cast<const CounterTopic*>(sample.get());
-            std::cout << "received: " << std::dec << data->counter << std::endl;
-            subscriber->release(userPayload);
-        });
-    }
-    else if (notification->getNotificationId() == SECOND_GROUP_ID)
+while (keepRunning)
+{
+    auto notificationVector = waitset.wait();
+
+    for (auto& notification : notificationVector)
```

Comment:
Was it intentional to have the while-loop also displayed or just a copy&past from the snippet above?
I'm just curious and fine either way.

Reply:
Mmmhh I underestimated your wisdom and geoffreys might - I thought you had a typo in your suggestion.